### ISSUE TITLE
[BUG] DEV9 - Random (?) Crashes

### ISSUE THREAD

**Issue by TheMrIron2** [PCSX2/pcsx2#4467]
**Describe the bug**

With DEV9 Ethernet enabled, multiple tested games seem to crash when they would not before - even those that don't use it.
(Games tested: Metal Gear Solid 3: Subsistence Disc 1 and Disc 2 (single/multiplayer), Burnout Revenge (offline singleplayer)
Completed a playthrough of MGS3 with identical settings (except for enabling speedhacks, which worked fine) without issues but issues appeared to occur when DEV9 was enabled, without any apparent pattern or reason. Similar crashes occurring to a friend on the same build.

**To Reproduce**

Due to the unpredictable nature of these crashes all I can suggest is to use my settings and see what happens.

**Expected behavior**

Normal gameplay.

**GS Settings**

https://i.imgur.com/AF5qNDp.png

**Emulation Settings**

All default. Speedhacks off, as visible: https://i.imgur.com/tebe3Zv.png
Additionally, the DEV9 window with Ethernet enabled and adapter selected: https://i.imgur.com/CRQOwms.png

**GS Window Screenshots**

N/A 

**System Info (please complete the following information):**
 -   PCSX2 Revision:  1.7.0-dev-1280
 -   OS:  Windows 10
 -   CPU:  i9-9880H
 -   GPU:  Radeon Pro 5500M

**Logs and Dumps**

Attached are two random logs from when I caught the game crashing. I can't see anything in common between them nor any errors - it seems the emulator hard-crashes before log output occurs.
[emuLog-bugged.txt](https://github.com/PCSX2/pcsx2/files/6563250/emuLog-bugged.txt)
[emuLog-bugged2.txt](https://github.com/PCSX2/pcsx2/files/6563251/emuLog-bugged2.txt)

Help is appreciated. Thank you.

**Comment by lightningterror** [comment]
Can you try older builds and see if you still experience crashing, could also be a more recent commit.

**Comment by TheMrIron2** [comment]
Hi, thanks for the quick response. Can confirm these same settings cause a similar crash on dev-1261, the oldest 1.7.0-dev I can currently access on-hand with the buildbot. Log attached.
[emuLog.txt](https://github.com/PCSX2/pcsx2/files/6563320/emuLog.txt)

**Comment by lightningterror** [comment]
> Hi, thanks for the quick response. Can confirm these same settings cause a similar crash on dev-1261, the oldest 1.7.0-dev I can currently access on-hand with the buildbot. Log attached.
> [emuLog.txt](https://github.com/PCSX2/pcsx2/files/6563320/emuLog.txt)

```
Bios Found: Europe  v01.60(04/10/2001)  Console
Bios Found: Europe  v02.00(14/06/2004)  Console
Bios Found: Japan   v01.00(17/01/2000)  Console
Bios Found: USA     v01.60(07/02/2002)  Console
```
Sorry but we don't support piracy, if you do own the console you'll have to send proof (pic of the console along with your username) and delete any bioses you manually downloaded and use/dump your own.

**Comment by TheMrIron2** [comment]
It seems there has been a misunderstanding. All of the consoles listed are, or were, indeed my own.
I live in Europe, so the earliest of those was my first model. I later bought an original Japanese and an NTSC PS2 so I could play imported discs on original hardware. My original 2001 console stopped reading discs about 10 years ago, so I sold it for parts and replaced it with the second Europe console on the list.
As a result, the only one still in my possession is the 2004 model (and is indeed the one I use for PCSX2), photo attached. The other three were sold at various points 5-10 years ago, but I dump data from all of my consoles - including PS2 - and I kept their dumps around because those consoles had a lot of memories for me and I didn't want to delete them if there was no need to.

If they are problematic I guess I can delete the others as they only have sentimental value, but the Europe one is legit and the pic below is that PS2 currently unplugged as I was installing a SATA mod on it recently, with username as requested. Forgive the confusion.

https://i.imgur.com/EZlrEr6.jpg

**Comment by lightningterror** [comment]
> Hi, thanks for the quick response. Can confirm these same settings cause a similar crash on dev-1261, the oldest 1.7.0-dev I can currently access on-hand with the buildbot. Log attached.

You need to view the full list which will show more builds.
https://buildbot.orphis.net/pcsx2/index.php?m=fulllist

**Comment by amneris1993** [comment]
Hi, I'm having the same problem that the game is crashing constantly. I'm trying different builds, but haven't had any luck even connecting to a game.

**Comment by TheMrIron2** [comment]
Hi, I tried dev-1263 (prior to a number of recent DEV9 changes) and crashes persisted. Even a version about 100 revisions prior (dev-1163) had the same crashes, although I expected as much given that few DEV9 changes occurred between 1263 and 1163.

Is there any particular build you would like me to test? It seems Amneris is having the same issue and I have had one or two other people echo the same complaint.
I'm in a community playing MGS3 online through PCSX2, so we all have similar setups, and all of us report these crashes with DEV9 enabled and not with it disabled - although frequency of crashes seems to vary for unknown reasons.

**Comment by TheLastRar** [comment]
Are all those who are reporting crashes using PCAP (Bridged)?

**Comment by TheMrIron2** [comment]
Yes, I believe so. The other people experiencing this issue followed the instructions I laid out to play MGS3 online, which I will post in full in case there is something else of note. But if they followed this - which I believe everyone I've talked to experiencing this issue - then everyone experiencing the crashes should be using PCAP (Bridged).
```
1. Install NPCAP https://nmap.org/npcap/
2. Go to PCSX2 -> Config -> Network and HDD Settings and enable Ethernet - select "PCAP (Bridged): WiFi" or a similarly named option. (Use "ipconfig" in command prompt to check the name if in doubt)
3. Download the MGS3 patch (https://cdn.discordapp.com/attachments/334473997838319626/335600801752023041/01B2FA7F.pnach) and place it in your PCSX2 cheats folder (usually Documents/PCSX2/cheats). Make sure cheats are enabled in System -> Game Settings.
(Optional) 4. Follow one of the methods in this short guide to set a static IP address: https://pureinfotech.com/set-static-ip-address-windows-10/.
5. In MGS3 Disc 2, go to ONLINE MODE and set up a network connection.
5a. Set the network device as SCE/Ethernet, then press Triangle for advanced settings and ensure the 100 Full Duplex option is selected.
5b. Set IP address manually and use your computer's IP, gateway and default router - type "ipconfig" into command prompt if unsure.
5c. Set DNS manually and use the SaveMGO DNSes (Primary: 192.3.217.61; Secondary 192.3.217.162)
```

**Comment by TheLastRar** [comment]
I had, at one point attempted to reproduce the issue, but wasn't successful at the time.

I did however, spot a potential issue which could have the caused crashes, can you check the actions build in https://github.com/PCSX2/pcsx2/pull/4496 to see if this issue is the same one you have?

**Comment by TheMrIron2** [comment]
Hi @TheLastRar, thanks for the update. Your new fix in #4496 seems to have eliminated the crashes. All of us report that the emulator is now stable and the previous random crashes are fixed.

Thanks again for looking into this.

**Comment by TheLastRar** [comment]
Thank you for testing
